x/feedistribution/keeper/allocation.go

```diff
@@ -0,0 +1,151 @@
+package keeper
+
+import (
+	"cosmossdk.io/math"
+	avstypes "github.com/ExocoreNetwork/exocore/x/avs/types"
+	"github.com/ExocoreNetwork/exocore/x/feedistribution/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+)
+
+// Based on the epoch, AllocateTokens performs reward and fee distribution to all validators.
+func (k Keeper) AllocateTokens(ctx sdk.Context, totalPreviousPower int64) error {
```

Comment:
This totalPreviousPower got from staking/dogfood's lateTotalPower is not actually previous power, it's actually the total power for next epoch, so as the validatorset got from Getallvalidators.
So that means we are distribution the reward from epoch_1 to validatorSet for epoch_2.
example:
1. epoch_1 has validatorSet{a:10,b:90}, totalPower100, received fee of 100exo as reward, so b should take 90(ignore tax here),
2.  but under current implementation ,we will use 'next power info' to do the distribution. Say at end of epoch_1 staking/dogfood update the powers to validatoSet(a:90, c:10), then b will got 0 reward for epoch_1 which is unacceptable.